engines/myst3/myst3.cpp

```diff
@@ -1877,6 +1900,17 @@ void Myst3Engine::pauseEngineIntern(bool pause) {
 
 	_state->pauseEngine(pause);
 
+	// Grab a game screen thumbnail in case we need one when writing a save file
+	// Get a thumbnail of the game screen
+	
+
+	if (_state->getLocationRoom() != 901) {
+		if (pause) {
```

Comment:
You should also be able to free the thumbnail memory if the game menu is not open when leaving the GMM.

Reply:
I am having the Menu object manage the thumbnails memory. 

If a new thumbnail is generated (generateSaveThumbnail() is called) then a reset(new thumbnail) is done in menu which deletes the old and replaces the new one. When the program exits that scoped pointer deletes the thumbnail if it is present in the Menu object. This is what I have intended to do at least.

I would prefer to not have to manage the logic for deletion of the thumbnail outside of the Menu class.

Is that sufficient?